cpp/inc/bond/comm/transport/epoxy.h

```diff
@@ -944,81 +913,133 @@ class EpoxyTransport
     class EpoxyServer
         : public INetworkServer
     {
-        typedef std::map<SocketAddress, boost::shared_ptr<INetworkConnectionSink>> SinkMap;
-
         //
-        // Schedule next incoming connection acceptance.
+        // Acceptor logic
         //
-        void DoAccept()
+        class EpoxyServerContext : public boost::enable_shared_from_this<EpoxyServerContext>
         {
-            m_acceptor.async_accept(
-                m_socket,
-                [this](const boost::system::error_code& ec) {
-                    if (!ec)
-                    {
-                        const boost::asio::ip::tcp::endpoint endpoint =  m_socket.remote_endpoint();
-                        const SocketAddress address(endpoint);
+            void OnAccept()
+            {
+                const boost::asio::ip::tcp::endpoint endpoint = m_socket.remote_endpoint();
+                const SocketAddress address(endpoint);
```

Comment:
Minor: Can we just pass `m_socket.remote_endpoint()` directly as an argument to `SocketAddress::SocketAddress`? We're not using `endpoint` elsewhere in this file and the expression is small enough that I don't think it needs a name.